dpctl/tensor/libtensor/include/kernels/accumulators.hpp

```diff
@@ -1255,20 +1222,19 @@ std::size_t cumsum_val_contig_impl(sycl::queue &q,
     }
     cumsumT *last_elem = cumsum_data_ptr + (n_elems - 1);
 
-    cumsumT *last_elem_host_usm = sycl::malloc_host<cumsumT>(1, q);
+    auto host_usm_owner =
+        dpctl::tensor::alloc_utils::smart_malloc_host<cumsumT>(1, q);
+    cumsumT *last_elem_host_usm = host_usm_owner.get();
 
-    if (last_elem_host_usm == nullptr) {
-        throw std::bad_alloc();
-    }
     sycl::event copy_e = q.submit([&](sycl::handler &cgh) {
         cgh.depends_on(comp_ev);
         cgh.copy<cumsumT>(last_elem, last_elem_host_usm, 1);
     });
     copy_e.wait();
     std::size_t return_val = static_cast<std::size_t>(*last_elem_host_usm);
 
-    using dpctl::tensor::alloc_utils::sycl_free_noexcept;
-    sycl_free_noexcept(last_elem_host_usm, q);
+    // free USM host allocation
+    host_usm_owner.reset(nullptr);
```

Comment:
You actually could call `reset` without parameters:
```c++
host_usm_owner.reset();
```

Also I'm not quite sure why you need to call it here at all.

Reply:
This was added to make it explicit that USM-host allocation was being freed. 

If removed, it would be done implicitly. Doing in it explicitly improves readability and maintainability, in my opinion.
I added a comment to make that intent clear.